testutils/src/main/scala/za/co/absa/enceladus/testutils/infoFileComparison/AtumModelUtils.scala

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright 2018-2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.testutils.infoFileComparison
+
+import org.apache.log4j.{LogManager, Logger}
+import za.co.absa.atum.model.{Checkpoint, ControlMeasure, ControlMeasureMetadata, Measurement}
+import za.co.absa.atum.utils.ControlUtils
+
+/**
+  *
+  * @param path Path to the difference through the model
+  * @param was Value it had in reference
+  * @param is Value it has now
+  * @tparam T Type of values
+  */
+case class ModelDifference[T](path: String, was: T, is: T)
+
+/**
+  * This object is used for [[ModelDifference]] object serialization
+  */
+object ModelDifferenceParser {
+  /**
+    * The method returns JSON representation of a [[ModelDifference]] object
+    */
+  def asJson(modelDifference: List[ModelDifference[_]]): String = {
+    ControlUtils.asJsonPretty[List[ModelDifference[_]]](modelDifference)
+  }
+}
+
+/**
+  * Object holding extensions of Atum Models for comparison purposes.
+  */
+object AtumModelUtils {
+  private val logger: Logger = LogManager.getLogger(this.getClass)
+
+  /**
+    * ControlMeasure's class extension adding compareWith
+    * @param controlMeasure Control Measure instance
+    */
+  implicit class ControlMeasureOps (controlMeasure: ControlMeasure){
+    /**
+      * Compare this ControLMeasure with the one passed in
+      * @param secondControlMeasure Second control measure
+      * @return Returns a list of model differences
+      */
+    def compareWith(secondControlMeasure: ControlMeasure): List[ModelDifference[_]] = {
+      val metadataDifferences = controlMeasure.metadata.compareWith(secondControlMeasure.metadata, "metadata")
+      val checkpointsDifferences = controlMeasure.checkpoints.zipWithIndex.foldLeft(List[ModelDifference[_]]()) {
+        case (agg, (value, index)) =>
```

Comment:
```suggestion
        case (agg, (checkpoint, index)) =>
```
I think calling it `checkpoint` might improve readability a little bit.

Reply:
make it a compromise? `checkpointValue`?
`checkpoint` on its own could be misleading when it is only some string that it is going to be compared to another.